Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -27,9 +63,82 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Frontend' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Widget' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+            'Enlight_Controller_Action_PreDispatch_Frontend_Checkout' => 'writeDefaultPaymentToSession',
+            'Shopware_Modules_Admin_UpdatePayment_FilterSql' => 'onApplePayButtonChangePaymentMethod',
         ];
     }
 
+    public function writeDefaultPaymentToSession(\Enlight_Event_EventArgs $args)
+    {
+        if (
+            !\in_array(
+                $this->adminModule->sGetUserData()['additional']['payment']['name'],
+                [self::APPLEPAY_NAME, self::APPLEPAYDIRECT_NAME],
+                true
+            )
+        ) {
+            return;
+        }
+
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+
+        if ($request->getActionName() !== 'finish') {
+            return;
+        }
+
+        $userId = $this->adminModule->sGetUserData()['additional']['user']['id'];
+
+        $customer = $this->modelManager->find(Customer::class, $userId);
+
+        $sql = $this->modelManager->getConnection()->prepare(
+            'SELECT paymentID FROM s_order WHERE userID = :userId AND paymentID NOT IN (
+                SELECT id FROM s_core_paymentmeans WHERE name = :applePayName OR name = :appleDirectName
+            ) ORDER BY ordertime DESC LIMIT 1;
+        ');
+
+        $sql->execute([
+            'userId' => $userId,
+            'applePayName' => self::APPLEPAY_NAME,
+            'appleDirectName' => self::APPLEPAYDIRECT_NAME
+        ]);
+
+        $result = $sql->fetch();
+
+        if (!is_array($result) && !\array_key_exists('paymentID', $result)) {
+            $paymentId = $this->config->get('defaultpayment');
+        } else {
+            $paymentId = $result['paymentID'];
+        }
+
+        $customer->setPaymentId($paymentId);
+
+        $this->modelManager->persist($customer);
+        $this->modelManager->flush();
+    }
+
+    public function onApplePayButtonChangePaymentMethod(\Enlight_Event_EventArgs $args)
```

Comment:
as described above, regarding to the event, this is more like the onUpdatePaymentMethod?

Reply:
Done